code/modules/hydroponics/biogenerator.dm

```diff
@@ -16,13 +16,17 @@
 	var/productivity = 0
 	var/max_items = 40
 	var/datum/techweb/stored_research
-	var/list/show_categories = list("Food", "Botany Chemicals", "Organic Materials")
+	var/bio_gen_part_tier
+	var/list/show_categories = list("Food", "Botany Chemicals", "Organic Materials", "Clothing")
 	/// Currently selected category in the UI
 	var/selected_cat
 
 /obj/machinery/biogenerator/Initialize()
 	. = ..()
-	stored_research = new /datum/techweb/specialized/autounlocking/biogenerator
+	for(var/obj/item/stock_parts/manipulator/M in component_parts)
+		var/datum/techweb/specialized/autounlocking/biogenerator/bio_parts = new
+		bio_parts.bio_gen_part_tier = M.rating
+		stored_research = bio_parts
```

Comment:
This would loop through all components, and overrides the last one's rating. Is this intentional?

Reply:
There only one manipulator in the biogenerator, so it only has one to check.
I believe this same method of part checking is used for the gene scanner as well.

Reply:
Ah, having a for loop just feels weird, since they normally loop through _multiple_ things